### DISCUSSION TITLE
Why my segments has different index type?

### DISCUSSION THREAD

**Discussion by scipe** [milvus-io/milvus#5234] [Q&A and General discussion] [answered]
Hi. I have a collection with partitions.
- dimension: 512
- index_file_size: 1024
- metric_type: IP
- index_type: IVF_SQ8
- index_nlist: 40000

And everything is ok, but for some reason when I click on collection -> any partition -> segments I see next:
![Screenshot 2021-05-14 at 14 30 53](https://user-images.githubusercontent.com/3996622/118265205-7a641800-b4c1-11eb-9b91-d92821d10986.png)

Why only one segment has IVF_SQ8 and all other has IDMAP?

**Comment by scipe** [comment, marked as answer]
Ok I found the answer: 
> Assume a collection has 10 segments, that means there are 10 raw files under the collection, and user create sq8 index, milvus will build index and generate 10 sq8 index files under the collection. So, there are 20 data files under this collection, 10 raw files, 10 index files. When user call show_table_info, the api will get all data files information and return. For raw file, its index type is "IDMAP", for index file, its index type is "SQ8"

**Reply by yhmo** [reply]
Could you tell me where the answer come from? I think there is a typo, the "show_table_info" should be "get_collection_info".

**Reply by scipe** [reply]
@yhmo sure mate. here: https://github.com/milvus-io/milvus/issues/1950#issuecomment-614610395

**Reply by yhmo** [reply]
The frontend call get_collection_stats() to get segment list from milvus. In old version, this api is named as "show_table_info", now it is renamed.
In old version, the show_table_info() list all segment files, including raw file and index file. But after 10.x, the get_collection_stats() behavior is changed: for segment that has finished index, the api return the index file info(index type is IVFSQ8); for segment that not yet finish index, the api return the raw file info(index type is IDMAP).

**Reply by scipe** [reply]
@yhmo is any way to see index progress status? Like in percents.. or maybe ETA.

**Reply by yhmo** [reply]
There is an issue [#3690](https://github.com/milvus-io/milvus/issues/3690) for this, but we didn't implement it.
